src/test/crossFetchNode.test.ts

```diff
@@ -274,3 +274,58 @@ test('should not fail when arbitrary parameters are configured in fetchOptions',
   });
   expect(response).toEqual({});
 });
+
+test('throws on responses other than and 304 errors', async () => {
+  nock('https://localhost:3000')
+    .get(
+      `/search/shopper-search/v1/organizations/${config.parameters.organizationId}/product-search?siteId=${config.parameters.siteId}&q=sony`
+    )
+    .matchHeader('authorization', 'Bearer test-auth')
+    .reply(400, {}, {'content-type': 'application-json charset=UTF-8'});
+
+  const standardConfig: ClientConfigInit<TestConfigParameters> = {...config};
+  const client = new ShopperSearch(standardConfig);
+  const response = async () =>
+    client.productSearch({
+      parameters: {q: 'sony'},
+      headers: {authorization: 'Bearer test-auth'},
+    });
```

Comment:
```suggestion
  const responsePromise = client.productSearch({
      parameters: {q: 'sony'},
      headers: {authorization: 'Bearer test-auth'},
    });
```
The API call is an async function, so even though you added a `throw` statement, it doesn't throw. It returns a promise that gets rejected. You don't need to wrap the call to test it.